service/worker/scheduler/workflow.go

```diff
@@ -1337,6 +1337,7 @@ func (s *scheduler) startWorkflow(
 			Header:                   newWorkflow.Header,
 			LastCompletionResult:     lastCompletionResult,
 			ContinuedFailure:         continuedFailure,
+			UserMetadata:             newWorkflow.GetUserMetadata(),
```

Comment:
you can just do
```suggestion
			UserMetadata:             newWorkflow.UserMetadata,
```
for consistency. if newWorkflow was nil we would already have panicked